### ISSUE TITLE
Refactor security enabled tests

### ISSUE THREAD

**Issue by jmazanec15** [opensearch-project/k-NN#901]
Security enabled tests are flaky. This is partially because they rely on the distro build and partially because the docker setup is somewhat complex.

See https://github.com/opensearch-project/k-NN/actions/workflows/test_security.yml for the failures.

I think to unblock our CI, we should remove this test and replace it by directly depending on artifacts produced from security's local maven publish: https://github.com/opensearch-project/security/blob/main/.github/workflows/maven-publish.yml. We can then make security enabled tasks a dedicated gradle task and avoid the docker setup.

**Comment by heemin32** [comment]
Can we prioritize to fix it rather than disable it unless it fails consistently?

**Comment by martin-gaievski** [comment]
I'm assuming that setting up a cluster with security plugin as part of the gradle task may be a tricky one, somewhat comparable with setting up a docker.  
I've checked several recent failed runs, seems there is a mix of errors, like build issues, incompatible changes from core upstream, some failed tests, etc. I think majority of this will fail with a proposed approach as well as we probably need to build from sources and then run tests, so I'm not sure we're addressing flakiness. I assume with a gradle task we'll get better error messages and address problems faster in case of legitimate failed tests. 

My suggestion is to create draft version of PR with a gradle task to check how much it can improve troubleshooting and how easier the setup of a test cluster is. It will give data points and make decision easier.

**Comment by jmazanec15** [comment]
The major difference is that docker depends on the distribution change which depends on all of the artifacts in the distribution. So, when a large breaking change in core breaks a lot of plugins in the distribution, the distribution will be blocked until all plugins are fixed. So, it creates kind of a circular dependency. Additionally, with docker, we dont get access to the error messages which makes trouble shooting harder as well.

Instead, we could independently install the security plugin (similar to what we do in neural search: https://github.com/opensearch-project/neural-search/blob/main/build.gradle#L221). And then have a dedicated gradle secure test task.

In the short term, I was thinking we could either remove the current task, or we could just ignore the failures.

**Comment by martin-gaievski** [comment]
Thanks Jack, those are valid points, just one remark - we can still merge our PR is needed, even if check in secure cluster is failing. One thing that I think is in favor of existing approach is that it's more like in a core pipeline because we're testing on a cluster with most of plugins, while with a new task we'll be checking only k-NN + security plugin. This can potentially produce inaccurate results.
I think we can ignore failures in CI for now

**Comment by jmazanec15** [comment]
Right, but I think our integration tests are run periodically at the distribution level by jenkins  (I think this was enabled recently). This does have the downside of us not catching it until after the code is checked in, but I think cases where other, unrelated plugins cause failures are much more rare, so I dont think it'll be a common case.

**Comment by heemin32** [comment]
I kind of agree with @jmazanec15 that there won't be much issue which will come from a difference between test environment in core release pipeline and the gradle approach as knn does not depends on those other plugins and most of current test failures are not from the knn functionality but issue with running the cluster itself. Isolating those build issue will help us to identify and fix the actual issue in knn easier and quicker. Especially, as we are doing same thing in neural search already, I have more confidence on gradle approach.
